presto-parquet/src/main/java/com/facebook/presto/parquet/cache/MetadataReader.java

```diff
@@ -273,7 +284,7 @@ private static ColumnMetaData decryptMetadata(RowGroup rowGroup, ColumnCryptoMet
         try {
             return Util.readColumnMetaData(tempInputStream, columnDecryptionSetup.getMetaDataDecryptor(), columnMetaDataAAD);
         }
-        catch (IOException e) {
+        catch (Exception e) {
```

Comment:
It might be not a good idea to cache all the exceptions, is it possible to cache the exception more explicitly?

Reply:
That is true. Change it back.